internal/spoke_k8s_client/spoke_k8s_client_manager.go

```diff
@@ -0,0 +1,44 @@
+package spoke_k8s_client
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpokeK8sClientManager maintains a set of spoke clients
+
+//go:generate mockgen --build_flags=--mod=mod -package=spoke_k8s_client -destination=mock_spoke_k8s_client_manager.go . SpokeK8sClientManager
+type SpokeK8sClientManager interface {
+	AddFromSecret(secret *corev1.Secret) (SpokeK8sClient, error)
```

Comment:
Is it worth considering making this a generic client cache rather than one that requires the caller create the client from a kubeconfig stored in a secret?

I could see us taking just the kubeconfig as input and using the hostname as a cache key, for example.

Reply:
If it's not worth the effort, maybe this belongs in the controller package rather than here, which is meant to be general purpose.

Reply:
yeah, just moved it the controllers package